persistence/src/test/java/de/cyface/persistence/PersistenceLayerTest.java

```diff
@@ -0,0 +1,152 @@
+        final float p0 = SensorManager.PRESSURE_STANDARD_ATMOSPHERE;
+        // Act
+        final float pressureAtSeaLevel = pressure(0., p0);
+        final float pressureAboveSeaLevel = pressure(10_000., p0);
+        final float pressureBelowSeaLevel = pressure(-500., p0);
+        // Assert
+        assertThat(pressureAtSeaLevel, is(equalTo(p0)));
+        assertThat(pressureAboveSeaLevel, is(equalTo(264.41486f)));
+        assertThat(pressureBelowSeaLevel, is(equalTo(1074.7656f)));
+    }
+
+    @Test
+    public void testLoadAscendFromPressures() {
+
+        // Arrange
+        final float p0 = SensorManager.PRESSURE_STANDARD_ATMOSPHERE;
+        final TrackV6 trackV6 = new TrackV6();
+        trackV6.addPressure(new Pressure(1L, pressure(0., p0)));
+        trackV6.addPressure(new Pressure(2L, pressure(1., p0)));
+        trackV6.addPressure(new Pressure(3L, pressure(-1., p0)));
+        trackV6.addPressure(new Pressure(4L, pressure(0., p0)));
+        trackV6.addPressure(new Pressure(5L, pressure(1., p0)));
+        trackV6.addPressure(new Pressure(6L, pressure(3., p0)));
+        trackV6.addPressure(new Pressure(7L, pressure(1., p0)));
+        trackV6.addPressure(new Pressure(8L, pressure(2., p0)));
+
+        // Act
+        final Double ascend = oocut.ascendFromPressures(Collections.singletonList(trackV6));
+
+        // Assert
+        assertThat(ascend, is(closeTo(6., 0.01)));
+    }
+
+    @Test
+    public void testLoadAscendFromGnss() {
+
+        // Arrange
+        final TrackV6 trackV6 = new TrackV6();
+        trackV6.addLocation(new GeoLocationV6(1L, 0., 0., 0., 1., 5., 5.));
+        trackV6.addLocation(new GeoLocationV6(1L, 0., 0., 1., 1., 5., 5.));
+        trackV6.addLocation(new GeoLocationV6(1L, 0., 0., -1., 1., 5., 5.));
+        trackV6.addLocation(new GeoLocationV6(1L, 0., 0., 0., 1., 5., 5.));
+        trackV6.addLocation(new GeoLocationV6(1L, 0., 0., 1., 1., 5., 5.));
+        trackV6.addLocation(new GeoLocationV6(1L, 0., 0., 3., 1., 5., 5.));
+        trackV6.addLocation(new GeoLocationV6(1L, 0., 0., 1., 1., 5., 5.));
+        trackV6.addLocation(new GeoLocationV6(1L, 0., 0., 2., 1., 5., 5.));
+
+        // Act
+        final Double ascend = oocut.ascendFromGNSS(Collections.singletonList(trackV6));
+
+        // Assert
+        assertThat(ascend, is(closeTo(6., 0.01)));
+    }
+
+    /**
+     * Calculates the pressure expected for a specific altitude and weather condition.
+     * <p>
+     * Based on the formula from {@code android.hardware.SensorManager#getAltitude(float, float)}.
+     *
+     * @param altitude The altitude to calculate the pressure for in meters above sea level.
+     * @param p0 The pressure at sea level for the specific weather condition.
+     * @return The atmospheric pressure in hPa.
+     */
+    @SuppressWarnings("SameParameterValue")
+    private float pressure(double altitude, float p0) {
+        return p0 * (float)Math.pow(1.0f - altitude / 44330.0f, 5.255f);
+    }
```

Comment:
I resolved the formula which is used in `SensorManager.getAltitude(p, p0)` to get the pressure data I need to generate to test the pressure for specific border altitudes we support:
- 10_000 meters above sea level which is a bit more then the highest point on earth (mount everest)
- -500 meters below sea level which is a bit less than the lowest point on earth (death see).

[We don't support diving, flying and entering deep caves right now, which is fine I guess :)]

Reply:
On mount everest the pressure is roughly 265 hPa (with slight variations because of the condition of the atmosphere (weather).